### ISSUE TITLE
replace "validation" with "inference"

### ISSUE THREAD

**Issue by balhoff** [validatingrdf/validatingrdf.github.io#21]
In Section 3.2 "Understanding the RDF Validation Problem", subsection "Inference", it says:

> Validation workflows will likely perform validation both before and after validation.

Based on the context, I think this should be:

> Validation workflows will likely perform validation both before and after **inference**.

**Comment by labra** [comment]
You are right, thank's for spotting it. I keep the issue open until I repair it in the book.

**Comment by balhoff** [comment]
This may not be the best place to discuss this, but we (Gene Ontology Consortium) are exploring applications for ShEx now and trying to figure out best practice for shape constraints that have some parts that should be asserted and some that can be inferred. So I was paying particular attention to this passage. One example is a transitive property, e.g. part_of. In some data we want only one part_of relationship asserted for a node. But in the inferred graph it's okay that it is transitively part_of many other nodes. So we will check this only on the asserted graph. But for the same shape, we may want to say that it has to be of type biological_process. But it is okay that the asserted type will be a more specific subclass. In this case we need to check the inferred graph.

So we're thinking about best ways to coordinate this. It's not very satisfying to duplicate the shapes into two files, one with how we want the assertions to look, and one with what must be inferrable. It would be nice to mark a particular constraint as only used when some flag ('inference') is present... not sure how that might work.

**Comment by labra** [comment]
That's an interesting use case.  It looks to me as some kind of directives that could be toggled off in some cases and on in other cases.

We currently don't support that in the core ShEx language, but it may be some feature to add. 

One related feature that ShEx already has is machine readable annotations so some specialized ShEx processor could look at those annotations and ignore or use them . However, that behaviour would be out of the spec.

**Comment by labra** [comment]
I raised [this issue](https://github.com/shexSpec/shex/issues/94) with some hint about a possible solution to keep a single shape to be used before and after validation.

**Comment by balhoff** [comment]
Thanks very much!
